### ISSUE TITLE
Player specific permissions

### ISSUE THREAD

**Issue by PureTryOut** [cuberite/cuberite#3283]
I would like a way to have player specific permissions. For example I could have a player which has the moderator rank, and all the permission groups that belong to that rank. This moderator specifically is also allowed to restart the server or reload configs when needed. The other moderators however, are not allowed to do this.
Currently there is no way to accomplish this. All moderators have the exact same permissions all the time. I would like a way to add or remove permissions per player (remove as in not allowing something even though it is allowed by the players rank).

#### Possible use-case
In my case this will be used for my [plugin](https://github.com/linuxmasterrace/townvalds). Players can have town specific ranks like mayor, assistent, recruiter, etc. Based on this rank given to the player by the town mayor, I want to dynamically change the players permissions, no matter what permissions he/she has by it's rank.

**Comment by NiLSPACE** [comment]
While we're at it we could also add complete permission groups to specific players.

**Comment by PureTryOut** [comment]
That would be good as well yes. Basically the same implementation as with ranks, but then assigning them to individual players instead.

**Comment by fralix** [comment]
yeah this would be nice! I think it would make cuberite more dynamic if you could!

Not sure but on other servers there are plugins for this, I heard about PEX on a minecraft server (and I got banned for trying to use it ^^) - as far as I know this plugin includes your idea, adding permissions to specific users, with support of * to give someone all the permissions, e.g. the admin

**Comment by PureTryOut** [comment]
@fralix PEX is a plugin for Bukkit/Spigot. The built-in permission system of Bukkit is very bare, and PEX extends it and makes it functional. It indeed allows for permissions per player.

It also supports giving a player every permission of a certain type using the * character.
Example (using my own plugin), I could give somebody all town related permissions using `townvalds.town.*`, instead of having to manually specify them all (`townvalds.town.spawn`, etc). It would be good to add this feature as well, but that's for a different issue.

**Comment by Pokechu22** [comment]
Actually, `*` isn't a PEX-specific thing, and in fact `*` has no special meaning programmatically with regards to permissions.  It's just that you can make (as a plugin author) permissions that group other permissions (which are referred to as child permissions, but it doesn't force a tree structure).  Most plugins provide a `*` permission that grants all permissions on that tree, but it isn't required and isn't done automatically.  Implementing it that way gives more configurability since you can also define, say, a `townvalds.defaultbuilder` that grants basic permissions that builders would want.

Look at the [sample plugin.yml](http://wiki.bukkit.org/Plugin_YAML) for an example of how the `*` permission is set up internally.

**Comment by PureTryOut** [comment]
I realize it's not PEX-specific, but they implement it. It's just handy to be able to group permissions, and I don't think there is a way to do that yet in Cuberite. However, again, that's something for a different issue imo.

**Comment by NiLSPACE** [comment]
The asteriks sign is already supported in Cuberite. They are really useful together with Cuberite's restrictions. I can give a certain group `worldedit.*` and restrict things I don't want people to use like `worldedit.scripting.execute`
